src/GitHub.App/ViewModels/PullRequestDetailViewModel.cs

```diff
@@ -426,38 +413,60 @@ string GetBranchDisplayName(string targetBranchLabel)
 
         IObservable<Unit> DoCheckout(object unused)
         {
-            IObservable<Unit> operation = null;
-
-            switch (CheckoutMode)
+            return Observable.Defer(async () =>
             {
-                case CheckoutMode.NeedsPull:
-                    operation = pullRequestsService.Pull(repository);
-                    break;
-                case CheckoutMode.Fetch:
-                    operation = pullRequestsService
+                var localBranches = await pullRequestsService.GetLocalBranches(repository, Model).ToList();
+
+                if (localBranches.Count > 0)
```

Comment:
Actually, thinking further, I think it's best to leave this as-is in case _somehow_ we end up with >1 local branch created. Using `FirstOrDefaultAsync` would mean that the actual branch could never be found, meaning that another branch will be checked out each time the link is clicked, exacerbating the problem.